sqlglot/transforms.py

```diff
@@ -109,6 +111,16 @@ def eliminate_qualify(expression: exp.Expression) -> exp.Expression:
         select_candidates = exp.Window if expression.is_star else (exp.Window, exp.Column)
         for expr in qualify_filters.find_all(select_candidates):
             if isinstance(expr, exp.Window):
+                for column in expr.find_all(exp.Column):
+                    for select in expression.selects:
+                        if select.alias_or_name == column.name:
+                            if isinstance(select, exp.Func):
+                                column.replace(select)
+                            elif isinstance(select, exp.Alias) and isinstance(
+                                select.args.get("this"), exp.Func
+                            ):
+                                column.replace(select.args.get("this"))
```

Comment:
Shouldn't we replace all aliases here with their original expression? A couple examples that will be missed now, if I'm not mistaken:

- `SELECT name as new_name`
- `SELECT name || 'test' as new_name`

These won't necessarily be `exp.Func`

Reply:
You are right, in your first case, it is a column, I will update then.

The second case is a exp.Func because `||` will be parsed into exp.Concat I think. 

However, now I think the generated column will always covered by an alias, because without alias they can not reuse in other place. Tell me if I right then I will remove the non-alias part.

Reply:
No the alias is not required, eg. this is a valid BQ query:

```sql
with data as (
  select 1 as id, 'a' as name
  union all
  select 1 AS ID, 'b' as name
)
select
  id,
  name,
from data
qualify row_number() over (partition by id order by name desc) = 1
```

Reply:
@georgesittas in your case, `name` is a column from the `data` table, not generated one.

Reply:
Ah, so I guess we wouldn't encounter a problem for it then after the transformation runs? If that's the case then what you said about doing this replacement only for aliases makes sense.

Reply:
And if you do this, then you can add an additional filter in the dict I mentioned in my other comment so that only the aliased projections are included.